### PR TITLE
Create script to export surveys in Excel format for SurveyGizmo

### PR DESCRIPTION
**Pull request by flodolo**
Also add survey for Hello and Firefox 42 Beta+Release

### CODE REVIEW

**Comment by flodolo** [comment]
@TheoChevalier r?

Double checking locales against this would be great
http://help.surveygizmo.com/help/article/link/create-a-translated-survey#available-languages

File: scripts/export_surveys [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        $current_line = 2;
+        foreach ($reference_data['strings'] as $string_id => $string_value) {
+            // Export an empty cell if the string is not localized
+            if (LangManager::isStringLocalized($string_id, $locale_data, $reference_data)) {
+                $translation = Utils::cleanString($locale_data['strings'][$string_id]);
+            } else {
+                $translation = '';
+            }
+
+            $objPHPExcel->getActiveSheet()->setCellValue("A{$current_line}", array_pop($reference_data['comments'][$string_id]));
```
Comment (TheoChevalier) [inline scripts/export_surveys]: I think you want that instead here:

            $comment = isset($reference_data['comments'][$string_id][0]) ? $reference_data['comments'][$string_id][0] : '';
            $objPHPExcel->getActiveSheet()->setCellValue("A{$current_line}", $comment);

I couldn't manage to get comments exported with array_pop, got a lot of "PHP Warning:  array_pop() expects parameter 1 to be array, null given in /home/theo/moz/github/langchecker/scripts/export_surveys on line 76"

File: scripts/export_surveys [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    $todo = count($locale_analysis['Identical']) +
+            count($locale_analysis['Missing']) +
+            LangManager::countErrors($locale_analysis['errors']);
+    if ($todo == 0) {
+        // Load existing strings for this locale
+        $locale_data = LangManager::loadSource($current_website, $current_locale, $current_filename);
+
+        // Create new PHPExcel object
+        $objPHPExcel = new \PHPExcel();
+        $objPHPExcel->getActiveSheet()->setCellValue('A1', 'Key');
```
Comment (TheoChevalier) [inline scripts/export_surveys]: setCellValue('A1', 'Comment')
It's only a column for comments, right?

File: scripts/export_surveys (hunk 1 above)
Comment (flodolo) [inline scripts/export_surveys]: You're missing one fundamental point: I'm not making up the format, I need the [format](https://bugzilla.mozilla.org/show_bug.cgi?id=1194765#c1) that SurveyGizmo expects

Comment (flodolo) [inline scripts/export_surveys]: Also, each string must have a comment with the key, that's why I didn't bother checking for existence. Which file are you testing? The only one with the proper format is the hello survey.

Comment (TheoChevalier) [inline scripts/export_surveys]: I guess you didn't want to copy/paste this link :) Ok, I thought it was supposed to work with any .lang file, so I picked survey2.lang and added a few comments. Will try with the right one, but sure where survey_hello_fx42.lang is, I can't find it in the repo :/

**Comment by TheoChevalier** [comment]
Ok, script exporting correctly several files, can't see anything to add/remove in the locales list, looks like we're good to go :) +r

**Comment by flodolo** [comment]
Thanks, merging.

**Pull request merged**